apps/wallet-mobile/src/Search/SearchContext.tsx

```diff
@@ -93,15 +93,23 @@ export const useSearchOnNavBar = ({
   placeholder,
   title,
   noBack = false,
+  navigateBack,
 }: {
   placeholder: string
   title: string
   noBack?: boolean
+  navigateBack?(): boolean
 }) => {
   const navigation = useNavigation()
 
   const {search, visible, showSearch, hideSearch, clearSearch} = useSearch()
 
+  useEffect(() => {
+    if (!navigateBack) return
+    const subscription = BackHandler.addEventListener('hardwareBackPress', navigateBack)
```

Comment:
hey @MichalSzorad in that case, it should hideSearch and clearSearch, otherwise it will present a different behaviour. Tap back (close the search), Swipe back will be navigating back, but won't close and clear the search.